### ISSUE TITLE
Python's cv2.cuda.GpuMat assignTo is not modifying target variable

### ISSUE THREAD

**Issue by r2d3** [opencv/opencv#22908]
### System Information

OpenCV python version: 4.6.0-dev
Operating System / Platform: Ubuntu 20.04.5 LTS
Python version: 3.8.10
CUDA 11.4.4

### Detailed description

On IRC (#opencv), a user reported to me that `assignTo` on `cv.cuda.GpuMat` does not modify the target object.

`a.assignTo(b)` 

We could change this to work as resize which returns the modified object:

`b = cv.resize(a, (0,0), b, 2, 2)`

But after thinking, do we really need to expose `assignTo` to Python ? as the doc says "Provides a functional form of convertTo."

If you think that we should correct it, I have a patch to be able to write:

`b = a.assignTo(b)`

### Steps to reproduce

```python
import cv2 as cv
import numpy as np


a = cv.cuda.GpuMat()
a.upload(np.ones((1,1)))
b = a.clone()
b.setTo(0)
b.assignTo(a)
print(a.download(), b.download())
print(a.cudaPtr(), b.cudaPtr())

# output: [[1.]] [[0.]]
# expected: [[0.]] [[0.]]
# pointers expected to be the same
```

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by alalek** [comment]
Thank you for the report!

> `CV_WRAP void assignTo(GpuMat& m, int type = -1) const;`

There is missing `CV_OUT` or `CV_IN_OUT` modifier at least.


> do we really need to expose assignTo to Python

Added RFC label.
Looks like it is not used anymore (and its behavior doesn't look clear).

**Comment by r2d3** [comment]
Yes I first made a patch with CV_OUT but after some thought, I think it will be better to remove CV_WRAP
